File: main/20170316-240.py
class Solution(object):
    def searchMatrix(self, matrix, target):
        """
        :type matrix: List[List[int]]
        :type target: int
        :rtype: bool
        """
        if not matrix: return False

        line_size = len(matrix[0])
        columns = 0
        rows = len(matrix) - 1

        while (columns <= line_size-1) and (rows >= 0):
            value = matrix[rows][columns]

            if value == target:
                return True
            elif value < target:
                columns += 1
            else:
                rows -= 1

        return False

File: main/20170316-240.rb
# @param {Integer[][]} matrix
# @param {Integer} target
# @return {Boolean}
#
def search_matrix(matrix, target)
  return false if matrix.empty? || matrix.first.empty?

  line_size = matrix.first.size

  columns = 0
  rows = matrix.size - 1

  while (rows >= 0) && (columns <= line_size-1)
    value = matrix[rows][columns]
    return true if value == target

    value > target ? (rows -= 1) : (columns += 1)
  end

  return false
end